### ISSUE TITLE
[Camera] Android Camera Plugin startImageStream gets stuck. 

### ISSUE THREAD

**Issue by JcmeLs** [flutter/flutter#114862]
## Basic info
Development env：
1. Flutter version：3.3.5
2. Camera plugin: 0.10.0+4 (Tried all versions of 0.10.0)  

Device info:
1. Model:Xiaomi 10 Ultra
2. Android version:Android 12  

## Steps to Reproduce

1. Execute `flutter run` on the code sample
2. Call startImageStream after camera initialize

**Expected results:** 
Can see Camera preview normally.
**Actual results:** 
On some models(like Xiaomi 10 Ultra), the camera preview page gets stuck.And some devices work fine.I've seen similar issues with devices that seem to have Black Sharks in the issues.  
In order to rule out a single device abnormal factor. I found 2 Xiaomi 10 Ultra for testing, and they can all reproduce.
<details>
<summary>Code sample</summary>

The code is the same as the Camera plugin example, except that startimagestream is called after initialize.  
 https://github.com/flutter/plugins/blob/main/packages/camera/camera/example/lib/main.dart
```dart
      await cameraController.initialize();
      await cameraController.startImageStream((image) => null);
```

</details>

<details>
  <summary>Logs</summary>

```
2022-11-08 09:47:59.271 10339-10339/io.flutter.plugins.cameraexample D/CameraDevice-JV-0: waitUntilIdle: E. id = 0
2022-11-08 09:47:59.449 10339-10339/io.flutter.plugins.cameraexample D/CameraDevice-JV-0: waitUntilIdle: X
2022-11-08 09:47:59.823 10339-10339/io.flutter.plugins.cameraexample W/BpBinder: PerfMonitor binderTransact :  time=369ms interface=android.hardware.camera2.ICameraDeviceUser code=6
2022-11-08 09:47:59.831 10339-10339/io.flutter.plugins.cameraexample I/Camera: startPreviewWithImageStream
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample I/Camera: CameraCaptureSession onConfigured
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample I/Camera: Updating builder settings
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: ExposureLockFeature
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: ExposurePointFeature
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: ZoomLevelFeature
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: AutoFocusFeature
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: NoiseReductionFeature
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample I/Camera: updateNoiseReduction | currentSetting: fast
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: FocusPointFeature
2022-11-08 09:47:59.831 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: ResolutionFeature
2022-11-08 09:47:59.832 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: SensorOrientationFeature
2022-11-08 09:47:59.832 10339-10432/io.flutter.plugins.cameraexample W/s.cameraexampl: Long monitor contention with owner main (10339) at void android.hardware.camera2.impl.CameraDeviceImpl.waitUntilIdle()(CameraDeviceImpl.java:1424) waiters=0 in void android.hardware.camera2.impl.CameraDeviceImpl$4.run() for 559ms
2022-11-08 09:47:59.832 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: FlashFeature
2022-11-08 09:47:59.832 10339-10362/io.flutter.plugins.cameraexample W/s.cameraexampl: Long monitor contention with owner main (10339) at void android.hardware.camera2.impl.CameraDeviceImpl.waitUntilIdle()(CameraDeviceImpl.java:1424) waiters=1 in void android.hardware.camera2.impl.CameraDeviceImpl$CameraDeviceCallbacks.onResultReceived(android.hardware.camera2.impl.CameraMetadataNative, android.hardware.camera2.impl.CaptureResultExtras, android.hardware.camera2.impl.PhysicalCaptureResultInfo[]) for 551ms
2022-11-08 09:47:59.836 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: ExposureOffsetFeature
2022-11-08 09:47:59.837 10339-10899/io.flutter.plugins.cameraexample D/Camera: Updating builder with feature: FpsRangeFeature
2022-11-08 09:47:59.837 10339-10899/io.flutter.plugins.cameraexample I/Camera: refreshPreviewCaptureSession
2022-11-08 09:47:59.849 10339-10568/io.flutter.plugins.cameraexample I/Camera: CameraCaptureSession onClosed
2022-11-08 09:48:00.125 10339-10571/io.flutter.plugins.cameraexample W/CameraDevice-JV-0: Device error received, code 1, frame number 0, request ID -1, subseq ID 0
2022-11-08 09:48:00.126 10339-10432/io.flutter.plugins.cameraexample I/Camera: open | onError
2022-11-08 09:48:00.126 10339-10432/io.flutter.plugins.cameraexample I/Camera: close
2022-11-08 09:48:00.127 10339-10432/io.flutter.plugins.cameraexample D/CameraExtImplXiaoMi: releaseCameraDevice: 0
2022-11-08 09:48:00.128 10339-10432/io.flutter.plugins.cameraexample D/CameraDevice-JV-0: close: E. id = 0
2022-11-08 09:48:00.129 10339-11012/io.flutter.plugins.cameraexample D/CameraInjector: notify Powerkeeper in async thread..
2022-11-08 09:48:00.200 10339-10362/io.flutter.plugins.cameraexample W/CameraDevice-JV-0: Device error received, code 3, frame number 386, request ID -1, subseq ID 0
2022-11-08 09:48:00.285 10339-10571/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.285 10339-10571/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 2
2022-11-08 09:48:00.287 10339-10571/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.287 10339-10571/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 3
2022-11-08 09:48:00.288 10339-10571/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.288 10339-10571/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 4
2022-11-08 09:48:00.288 10339-10611/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.288 10339-10611/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 5
2022-11-08 09:48:00.288 10339-10611/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.289 10339-10611/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 6
2022-11-08 09:48:00.289 10339-10611/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.289 10339-10611/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 7
2022-11-08 09:48:00.289 10339-10749/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.289 10339-10749/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 9
2022-11-08 09:48:00.290 10339-10749/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.290 10339-10749/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 10
2022-11-08 09:48:00.290 10339-10749/io.flutter.plugins.cameraexample W/libc: Access denied finding property "vendor.camera.aux.packagelist"
2022-11-08 09:48:00.290 10339-10749/io.flutter.plugins.cameraexample W/CameraManagerGlobal: ignore the torch status update of camera: 11
2022-11-08 09:48:00.291 10339-10749/io.flutter.plugins.cameraexample E/CameraManagerGlobal: Camera 5 is not available. Ignore physical camera status change
2022-11-08 09:48:00.291 10339-10571/io.flutter.plugins.cameraexample E/CameraManagerGlobal: Camera 6 is not available. Ignore physical camera status change
2022-11-08 09:48:00.292 10339-10571/io.flutter.plugins.cameraexample E/CameraManagerGlobal: Camera 9 is not available. Ignore physical camera status change
2022-11-08 09:48:00.294 10339-10432/io.flutter.plugins.cameraexample D/CameraDevice-JV-0: close: X
2022-11-08 09:48:00.295 10339-10432/io.flutter.plugins.cameraexample I/Camera: The camera device has encountered a fatal error
2022-11-08 09:48:00.295 10339-10432/io.flutter.plugins.cameraexample I/Camera: open | onClosed
```
</details>

<details>
  <summary>Flutter doctor </summary>
```
[✓] Flutter (Channel stable, 3.3.5, on macOS 13.0 22A380 darwin-arm, locale en-CN)
    • Flutter version 3.3.5 on channel stable at /Users/liba2/Documents/work/flutter sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d9111f6402 (3 weeks ago), 2022-10-19 12:27:13 -0700
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/liba2/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Connected device (2 available)

```

</details>

**Comment by huycozy** [comment]
Hi @JcmeLs
I tried running [Android image streaming test](https://github.com/flutter/plugins/blob/main/packages/camera/camera/example/integration_test/camera_test.dart#L209-L244) which calls `startImageStream` after `initialize `, but can not reproduce this issue on my devices: Samsung Galaxy Tab A7 Lite, Realme 6. The test is passed.

Can you try running that test on your devices and check the test results? If the test is also passed, please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we may verify this. Thank you!

**Comment by JcmeLs** [comment]
hello @huycozy ，thank you for your reply.After testing, the Android image streaming test of integration_test can pass on my device. But I can't debug into the onLatestImageAvailable callback. In the actual use process, found that the first frame can be previewed normally, and it will be stuck after the first frame. And only on some models, like my Mi 10 Ultra. I have also tried different MIUI versions and this problem occurs too.
The complete code is as follows, just call startImageStream after initialize. There are no other changes.
``` dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'dart:async';
import 'dart:io';

import 'package:camera/camera.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/scheduler.dart';
import 'package:video_player/video_player.dart';

/// Camera example home widget.
class CameraExampleHome extends StatefulWidget {
  /// Default Constructor
  const CameraExampleHome({Key? key}) : super(key: key);

  @override
  State<CameraExampleHome> createState() {
    return _CameraExampleHomeState();
  }
}

/// Returns a suitable camera icon for [direction].
IconData getCameraLensIcon(CameraLensDirection direction) {
  switch (direction) {
    case CameraLensDirection.back:
      return Icons.camera_rear;
    case CameraLensDirection.front:
      return Icons.camera_front;
    case CameraLensDirection.external:
      return Icons.camera;
    default:
      throw ArgumentError('Unknown lens direction');
  }
}

void _logError(String code, String? message) {
  if (message != null) {
    print('Error: $code\nError Message: $message');
  } else {
    print('Error: $code');
  }
}

class _CameraExampleHomeState extends State<CameraExampleHome>
    with WidgetsBindingObserver, TickerProviderStateMixin {
  CameraController? controller;
  XFile? imageFile;
  XFile? videoFile;
  VideoPlayerController? videoController;
  VoidCallback? videoPlayerListener;
  bool enableAudio = true;
  double _minAvailableExposureOffset = 0.0;
  double _maxAvailableExposureOffset = 0.0;
  double _currentExposureOffset = 0.0;
  late AnimationController _flashModeControlRowAnimationController;
  late Animation<double> _flashModeControlRowAnimation;
  late AnimationController _exposureModeControlRowAnimationController;
  late Animation<double> _exposureModeControlRowAnimation;
  late AnimationController _focusModeControlRowAnimationController;
  late Animation<double> _focusModeControlRowAnimation;
  double _minAvailableZoom = 1.0;
  double _maxAvailableZoom = 1.0;
  double _currentScale = 1.0;
  double _baseScale = 1.0;

  // Counting pointers (number of user fingers on screen)
  int _pointers = 0;

  @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addObserver(this);

    _flashModeControlRowAnimationController = AnimationController(
      duration: const Duration(milliseconds: 300),
      vsync: this,
    );
    _flashModeControlRowAnimation = CurvedAnimation(
      parent: _flashModeControlRowAnimationController,
      curve: Curves.easeInCubic,
    );
    _exposureModeControlRowAnimationController = AnimationController(
      duration: const Duration(milliseconds: 300),
      vsync: this,
    );
    _exposureModeControlRowAnimation = CurvedAnimation(
      parent: _exposureModeControlRowAnimationController,
      curve: Curves.easeInCubic,
    );
    _focusModeControlRowAnimationController = AnimationController(
      duration: const Duration(milliseconds: 300),
      vsync: this,
    );
    _focusModeControlRowAnimation = CurvedAnimation(
      parent: _focusModeControlRowAnimationController,
      curve: Curves.easeInCubic,
    );
  }

  @override
  void dispose() {
    WidgetsBinding.instance.removeObserver(this);
    _flashModeControlRowAnimationController.dispose();
    _exposureModeControlRowAnimationController.dispose();
    super.dispose();
  }

  // #docregion AppLifecycle
  @override
  void didChangeAppLifecycleState(AppLifecycleState state) {
    final CameraController? cameraController = controller;

    // App state changed before we got the chance to initialize.
    if (cameraController == null || !cameraController.value.isInitialized) {
      return;
    }

    if (state == AppLifecycleState.inactive) {
      cameraController.dispose();
    } else if (state == AppLifecycleState.resumed) {
      onNewCameraSelected(cameraController.description);
    }
  }
  // #enddocregion AppLifecycle

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Camera example'),
      ),
      body: Column(
        children: <Widget>[
          Expanded(
            child: Container(
              decoration: BoxDecoration(
                color: Colors.black,
                border: Border.all(
                  color:
                  controller != null && controller!.value.isRecordingVideo
                      ? Colors.redAccent
                      : Colors.grey,
                  width: 3.0,
                ),
              ),
              child: Padding(
                padding: const EdgeInsets.all(1.0),
                child: Center(
                  child: _cameraPreviewWidget(),
                ),
              ),
            ),
          ),
          _captureControlRowWidget(),
          _modeControlRowWidget(),
          Padding(
            padding: const EdgeInsets.all(5.0),
            child: Row(
              children: <Widget>[
                _cameraTogglesRowWidget(),
                _thumbnailWidget(),
              ],
            ),
          ),
        ],
      ),
    );
  }

  /// Display the preview from the camera (or a message if the preview is not available).
  Widget _cameraPreviewWidget() {
    final CameraController? cameraController = controller;

    if (cameraController == null || !cameraController.value.isInitialized) {
      return const Text(
        'Tap a camera',
        style: TextStyle(
          color: Colors.white,
          fontSize: 24.0,
          fontWeight: FontWeight.w900,
        ),
      );
    } else {
      return Listener(
        onPointerDown: (_) => _pointers++,
        onPointerUp: (_) => _pointers--,
        child: CameraPreview(
          controller!,
          child: LayoutBuilder(
              builder: (BuildContext context, BoxConstraints constraints) {
                return GestureDetector(
                  behavior: HitTestBehavior.opaque,
                  onScaleStart: _handleScaleStart,
                  onScaleUpdate: _handleScaleUpdate,
                  onTapDown: (TapDownDetails details) =>
                      onViewFinderTap(details, constraints),
                );
              }),
        ),
      );
    }
  }

  void _handleScaleStart(ScaleStartDetails details) {
    _baseScale = _currentScale;
  }

  Future<void> _handleScaleUpdate(ScaleUpdateDetails details) async {
    // When there are not exactly two fingers on screen don't scale
    if (controller == null || _pointers != 2) {
      return;
    }

    _currentScale = (_baseScale * details.scale)
        .clamp(_minAvailableZoom, _maxAvailableZoom);

    await controller!.setZoomLevel(_currentScale);
  }

  /// Display the thumbnail of the captured image or video.
  Widget _thumbnailWidget() {
    final VideoPlayerController? localVideoController = videoController;

    return Expanded(
      child: Align(
        alignment: Alignment.centerRight,
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: <Widget>[
            if (localVideoController == null && imageFile == null)
              Container()
            else
              SizedBox(
                width: 64.0,
                height: 64.0,
                child: (localVideoController == null)
                    ? (
                    // The captured image on the web contains a network-accessible URL
                    // pointing to a location within the browser. It may be displayed
                    // either with Image.network or Image.memory after loading the image
                    // bytes to memory.
                    kIsWeb
                        ? Image.network(imageFile!.path)
                        : Image.file(File(imageFile!.path)))
                    : Container(
                  decoration: BoxDecoration(
                      border: Border.all(color: Colors.pink)),
                  child: Center(
                    child: AspectRatio(
                        aspectRatio:
                        localVideoController.value.size != null
                            ? localVideoController.value.aspectRatio
                            : 1.0,
                        child: VideoPlayer(localVideoController)),
                  ),
                ),
              ),
          ],
        ),
      ),
    );
  }

  /// Display a bar with buttons to change the flash and exposure modes
  Widget _modeControlRowWidget() {
    return Column(
      children: <Widget>[
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
          children: <Widget>[
            IconButton(
              icon: const Icon(Icons.flash_on),
              color: Colors.blue,
              onPressed: controller != null ? onFlashModeButtonPressed : null,
            ),
            // The exposure and focus mode are currently not supported on the web.
            ...!kIsWeb
                ? <Widget>[
              IconButton(
                icon: const Icon(Icons.exposure),
                color: Colors.blue,
                onPressed: controller != null
                    ? onExposureModeButtonPressed
                    : null,
              ),
              IconButton(
                icon: const Icon(Icons.filter_center_focus),
                color: Colors.blue,
                onPressed:
                controller != null ? onFocusModeButtonPressed : null,
              )
            ]
                : <Widget>[],
            IconButton(
              icon: Icon(enableAudio ? Icons.volume_up : Icons.volume_mute),
              color: Colors.blue,
              onPressed: controller != null ? onAudioModeButtonPressed : null,
            ),
            IconButton(
              icon: Icon(controller?.value.isCaptureOrientationLocked ?? false
                  ? Icons.screen_lock_rotation
                  : Icons.screen_rotation),
              color: Colors.blue,
              onPressed: controller != null
                  ? onCaptureOrientationLockButtonPressed
                  : null,
            ),
          ],
        ),
        _flashModeControlRowWidget(),
        _exposureModeControlRowWidget(),
        _focusModeControlRowWidget(),
      ],
    );
  }

  Widget _flashModeControlRowWidget() {
    return SizeTransition(
      sizeFactor: _flashModeControlRowAnimation,
      child: ClipRect(
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
          children: <Widget>[
            IconButton(
              icon: const Icon(Icons.flash_off),
              color: controller?.value.flashMode == FlashMode.off
                  ? Colors.orange
                  : Colors.blue,
              onPressed: controller != null
                  ? () => onSetFlashModeButtonPressed(FlashMode.off)
                  : null,
            ),
            IconButton(
              icon: const Icon(Icons.flash_auto),
              color: controller?.value.flashMode == FlashMode.auto
                  ? Colors.orange
                  : Colors.blue,
              onPressed: controller != null
                  ? () => onSetFlashModeButtonPressed(FlashMode.auto)
                  : null,
            ),
            IconButton(
              icon: const Icon(Icons.flash_on),
              color: controller?.value.flashMode == FlashMode.always
                  ? Colors.orange
                  : Colors.blue,
              onPressed: controller != null
                  ? () => onSetFlashModeButtonPressed(FlashMode.always)
                  : null,
            ),
            IconButton(
              icon: const Icon(Icons.highlight),
              color: controller?.value.flashMode == FlashMode.torch
                  ? Colors.orange
                  : Colors.blue,
              onPressed: controller != null
                  ? () => onSetFlashModeButtonPressed(FlashMode.torch)
                  : null,
            ),
          ],
        ),
      ),
    );
  }

  Widget _exposureModeControlRowWidget() {
    final ButtonStyle styleAuto = TextButton.styleFrom(
      // TODO(darrenaustin): Migrate to new API once it lands in stable: https://github.com/flutter/flutter/issues/105724
      // ignore: deprecated_member_use
      primary: controller?.value.exposureMode == ExposureMode.auto
          ? Colors.orange
          : Colors.blue,
    );
    final ButtonStyle styleLocked = TextButton.styleFrom(
      // TODO(darrenaustin): Migrate to new API once it lands in stable: https://github.com/flutter/flutter/issues/105724
      // ignore: deprecated_member_use
      primary: controller?.value.exposureMode == ExposureMode.locked
          ? Colors.orange
          : Colors.blue,
    );

    return SizeTransition(
      sizeFactor: _exposureModeControlRowAnimation,
      child: ClipRect(
        child: Container(
          color: Colors.grey.shade50,
          child: Column(
            children: <Widget>[
              const Center(
                child: Text('Exposure Mode'),
              ),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  TextButton(
                    style: styleAuto,
                    onPressed: controller != null
                        ? () =>
                        onSetExposureModeButtonPressed(ExposureMode.auto)
                        : null,
                    onLongPress: () {
                      if (controller != null) {
                        controller!.setExposurePoint(null);
                        showInSnackBar('Resetting exposure point');
                      }
                    },
                    child: const Text('AUTO'),
                  ),
                  TextButton(
                    style: styleLocked,
                    onPressed: controller != null
                        ? () =>
                        onSetExposureModeButtonPressed(ExposureMode.locked)
                        : null,
                    child: const Text('LOCKED'),
                  ),
                  TextButton(
                    style: styleLocked,
                    onPressed: controller != null
                        ? () => controller!.setExposureOffset(0.0)
                        : null,
                    child: const Text('RESET OFFSET'),
                  ),
                ],
              ),
              const Center(
                child: Text('Exposure Offset'),
              ),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  Text(_minAvailableExposureOffset.toString()),
                  Slider(
                    value: _currentExposureOffset,
                    min: _minAvailableExposureOffset,
                    max: _maxAvailableExposureOffset,
                    label: _currentExposureOffset.toString(),
                    onChanged: _minAvailableExposureOffset ==
                        _maxAvailableExposureOffset
                        ? null
                        : setExposureOffset,
                  ),
                  Text(_maxAvailableExposureOffset.toString()),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _focusModeControlRowWidget() {
    final ButtonStyle styleAuto = TextButton.styleFrom(
      // TODO(darrenaustin): Migrate to new API once it lands in stable: https://github.com/flutter/flutter/issues/105724
      // ignore: deprecated_member_use
      primary: controller?.value.focusMode == FocusMode.auto
          ? Colors.orange
          : Colors.blue,
    );
    final ButtonStyle styleLocked = TextButton.styleFrom(
      // TODO(darrenaustin): Migrate to new API once it lands in stable: https://github.com/flutter/flutter/issues/105724
      // ignore: deprecated_member_use
      primary: controller?.value.focusMode == FocusMode.locked
          ? Colors.orange
          : Colors.blue,
    );

    return SizeTransition(
      sizeFactor: _focusModeControlRowAnimation,
      child: ClipRect(
        child: Container(
          color: Colors.grey.shade50,
          child: Column(
            children: <Widget>[
              const Center(
                child: Text('Focus Mode'),
              ),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: <Widget>[
                  TextButton(
                    style: styleAuto,
                    onPressed: controller != null
                        ? () => onSetFocusModeButtonPressed(FocusMode.auto)
                        : null,
                    onLongPress: () {
                      if (controller != null) {
                        controller!.setFocusPoint(null);
                      }
                      showInSnackBar('Resetting focus point');
                    },
                    child: const Text('AUTO'),
                  ),
                  TextButton(
                    style: styleLocked,
                    onPressed: controller != null
                        ? () => onSetFocusModeButtonPressed(FocusMode.locked)
                        : null,
                    child: const Text('LOCKED'),
                  ),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }

  /// Display the control bar with buttons to take pictures and record videos.
  Widget _captureControlRowWidget() {
    final CameraController? cameraController = controller;

    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
      children: <Widget>[
        IconButton(
          icon: const Icon(Icons.camera_alt),
          color: Colors.blue,
          onPressed: cameraController != null &&
              cameraController.value.isInitialized &&
              !cameraController.value.isRecordingVideo
              ? onTakePictureButtonPressed
              : null,
        ),
        IconButton(
          icon: const Icon(Icons.videocam),
          color: Colors.blue,
          onPressed: cameraController != null &&
              cameraController.value.isInitialized &&
              !cameraController.value.isRecordingVideo
              ? onVideoRecordButtonPressed
              : null,
        ),
        IconButton(
          icon: cameraController != null &&
              cameraController.value.isRecordingPaused
              ? const Icon(Icons.play_arrow)
              : const Icon(Icons.pause),
          color: Colors.blue,
          onPressed: cameraController != null &&
              cameraController.value.isInitialized &&
              cameraController.value.isRecordingVideo
              ? (cameraController.value.isRecordingPaused)
              ? onResumeButtonPressed
              : onPauseButtonPressed
              : null,
        ),
        IconButton(
          icon: const Icon(Icons.stop),
          color: Colors.red,
          onPressed: cameraController != null &&
              cameraController.value.isInitialized &&
              cameraController.value.isRecordingVideo
              ? onStopButtonPressed
              : null,
        ),
        IconButton(
          icon: const Icon(Icons.pause_presentation),
          color:
          cameraController != null && cameraController.value.isPreviewPaused
              ? Colors.red
              : Colors.blue,
          onPressed:
          cameraController == null ? null : onPausePreviewButtonPressed,
        ),
      ],
    );
  }

  /// Display a row of toggle to select the camera (or a message if no camera is available).
  Widget _cameraTogglesRowWidget() {
    final List<Widget> toggles = <Widget>[];

    void onChanged(CameraDescription? description) {
      if (description == null) {
        return;
      }

      onNewCameraSelected(description);
    }

    if (_cameras.isEmpty) {
      SchedulerBinding.instance.addPostFrameCallback((_) async {
        showInSnackBar('No camera found.');
      });
      return const Text('None');
    } else {
      for (final CameraDescription cameraDescription in _cameras) {
        toggles.add(
          SizedBox(
            width: 90.0,
            child: RadioListTile<CameraDescription>(
              title: Icon(getCameraLensIcon(cameraDescription.lensDirection)),
              groupValue: controller?.description,
              value: cameraDescription,
              onChanged:
              controller != null && controller!.value.isRecordingVideo
                  ? null
                  : onChanged,
            ),
          ),
        );
      }
    }

    return Row(children: toggles);
  }

  String timestamp() => DateTime.now().millisecondsSinceEpoch.toString();

  void showInSnackBar(String message) {
    ScaffoldMessenger.of(context)
        .showSnackBar(SnackBar(content: Text(message)));
  }

  void onViewFinderTap(TapDownDetails details, BoxConstraints constraints) {
    if (controller == null) {
      return;
    }

    final CameraController cameraController = controller!;

    final Offset offset = Offset(
      details.localPosition.dx / constraints.maxWidth,
      details.localPosition.dy / constraints.maxHeight,
    );
    cameraController.setExposurePoint(offset);
    cameraController.setFocusPoint(offset);
  }

  Future<void> onNewCameraSelected(CameraDescription cameraDescription) async {
    final CameraController? oldController = controller;
    if (oldController != null) {
      // `controller` needs to be set to null before getting disposed,
      // to avoid a race condition when we use the controller that is being
      // disposed. This happens when camera permission dialog shows up,
      // which triggers `didChangeAppLifecycleState`, which disposes and
      // re-creates the controller.
      controller = null;
      await oldController.dispose();
    }

    final CameraController cameraController = CameraController(
      cameraDescription,
      kIsWeb ? ResolutionPreset.max : ResolutionPreset.medium,
      enableAudio: enableAudio,
      imageFormatGroup: ImageFormatGroup.jpeg,
    );

    controller = cameraController;

    // If the controller is updated then update the UI.
    cameraController.addListener(() {
      if (mounted) {
        setState(() {});
      }
      if (cameraController.value.hasError) {
        showInSnackBar(
            'Camera error ${cameraController.value.errorDescription}');
      }
    });

    try {
      await cameraController.initialize();
      await cameraController.startImageStream((image) => null);
      await Future.wait(<Future<Object?>>[
        // The exposure mode is currently not supported on the web.
        ...!kIsWeb
            ? <Future<Object?>>[
          cameraController.getMinExposureOffset().then(
                  (double value) => _minAvailableExposureOffset = value),
          cameraController
              .getMaxExposureOffset()
              .then((double value) => _maxAvailableExposureOffset = value)
        ]
            : <Future<Object?>>[],
        cameraController
            .getMaxZoomLevel()
            .then((double value) => _maxAvailableZoom = value),
        cameraController
            .getMinZoomLevel()
            .then((double value) => _minAvailableZoom = value),
      ]);
    } on CameraException catch (e) {
      switch (e.code) {
        case 'CameraAccessDenied':
          showInSnackBar('You have denied camera access.');
          break;
        case 'CameraAccessDeniedWithoutPrompt':
        // iOS only
          showInSnackBar('Please go to Settings app to enable camera access.');
          break;
        case 'CameraAccessRestricted':
        // iOS only
          showInSnackBar('Camera access is restricted.');
          break;
        case 'AudioAccessDenied':
          showInSnackBar('You have denied audio access.');
          break;
        case 'AudioAccessDeniedWithoutPrompt':
        // iOS only
          showInSnackBar('Please go to Settings app to enable audio access.');
          break;
        case 'AudioAccessRestricted':
        // iOS only
          showInSnackBar('Audio access is restricted.');
          break;
        default:
          _showCameraException(e);
          break;
      }
    }

    if (mounted) {
      setState(() {});
    }
  }

  void onTakePictureButtonPressed() {
    takePicture().then((XFile? file) {
      if (mounted) {
        setState(() {
          imageFile = file;
          videoController?.dispose();
          videoController = null;
        });
        if (file != null) {
          showInSnackBar('Picture saved to ${file.path}');
        }
      }
    });
  }

  void onFlashModeButtonPressed() {
    if (_flashModeControlRowAnimationController.value == 1) {
      _flashModeControlRowAnimationController.reverse();
    } else {
      _flashModeControlRowAnimationController.forward();
      _exposureModeControlRowAnimationController.reverse();
      _focusModeControlRowAnimationController.reverse();
    }
  }

  void onExposureModeButtonPressed() {
    if (_exposureModeControlRowAnimationController.value == 1) {
      _exposureModeControlRowAnimationController.reverse();
    } else {
      _exposureModeControlRowAnimationController.forward();
      _flashModeControlRowAnimationController.reverse();
      _focusModeControlRowAnimationController.reverse();
    }
  }

  void onFocusModeButtonPressed() {
    if (_focusModeControlRowAnimationController.value == 1) {
      _focusModeControlRowAnimationController.reverse();
    } else {
      _focusModeControlRowAnimationController.forward();
      _flashModeControlRowAnimationController.reverse();
      _exposureModeControlRowAnimationController.reverse();
    }
  }

  void onAudioModeButtonPressed() {
    enableAudio = !enableAudio;
    if (controller != null) {
      onNewCameraSelected(controller!.description);
    }
  }

  Future<void> onCaptureOrientationLockButtonPressed() async {
    try {
      if (controller != null) {
        final CameraController cameraController = controller!;
        if (cameraController.value.isCaptureOrientationLocked) {
          await cameraController.unlockCaptureOrientation();
          showInSnackBar('Capture orientation unlocked');
        } else {
          await cameraController.lockCaptureOrientation();
          showInSnackBar(
              'Capture orientation locked to ${cameraController.value.lockedCaptureOrientation.toString().split('.').last}');
        }
      }
    } on CameraException catch (e) {
      _showCameraException(e);
    }
  }

  void onSetFlashModeButtonPressed(FlashMode mode) {
    setFlashMode(mode).then((_) {
      if (mounted) {
        setState(() {});
      }
      showInSnackBar('Flash mode set to ${mode.toString().split('.').last}');
    });
  }

  void onSetExposureModeButtonPressed(ExposureMode mode) {
    setExposureMode(mode).then((_) {
      if (mounted) {
        setState(() {});
      }
      showInSnackBar('Exposure mode set to ${mode.toString().split('.').last}');
    });
  }

  void onSetFocusModeButtonPressed(FocusMode mode) {
    setFocusMode(mode).then((_) {
      if (mounted) {
        setState(() {});
      }
      showInSnackBar('Focus mode set to ${mode.toString().split('.').last}');
    });
  }

  void onVideoRecordButtonPressed() {
    startVideoRecording().then((_) {
      if (mounted) {
        setState(() {});
      }
    });
  }

  void onStopButtonPressed() {
    stopVideoRecording().then((XFile? file) {
      if (mounted) {
        setState(() {});
      }
      if (file != null) {
        showInSnackBar('Video recorded to ${file.path}');
        videoFile = file;
        _startVideoPlayer();
      }
    });
  }

  Future<void> onPausePreviewButtonPressed() async {
    final CameraController? cameraController = controller;

    if (cameraController == null || !cameraController.value.isInitialized) {
      showInSnackBar('Error: select a camera first.');
      return;
    }

    if (cameraController.value.isPreviewPaused) {
      await cameraController.resumePreview();
    } else {
      await cameraController.pausePreview();
    }

    if (mounted) {
      setState(() {});
    }
  }

  void onPauseButtonPressed() {
    pauseVideoRecording().then((_) {
      if (mounted) {
        setState(() {});
      }
      showInSnackBar('Video recording paused');
    });
  }

  void onResumeButtonPressed() {
    resumeVideoRecording().then((_) {
      if (mounted) {
        setState(() {});
      }
      showInSnackBar('Video recording resumed');
    });
  }

  Future<void> startVideoRecording() async {
    final CameraController? cameraController = controller;

    if (cameraController == null || !cameraController.value.isInitialized) {
      showInSnackBar('Error: select a camera first.');
      return;
    }

    if (cameraController.value.isRecordingVideo) {
      // A recording is already started, do nothing.
      return;
    }

    try {
      await cameraController.startVideoRecording();
    } on CameraException catch (e) {
      _showCameraException(e);
      return;
    }
  }

  Future<XFile?> stopVideoRecording() async {
    final CameraController? cameraController = controller;

    if (cameraController == null || !cameraController.value.isRecordingVideo) {
      return null;
    }

    try {
      return cameraController.stopVideoRecording();
    } on CameraException catch (e) {
      _showCameraException(e);
      return null;
    }
  }

  Future<void> pauseVideoRecording() async {
    final CameraController? cameraController = controller;

    if (cameraController == null || !cameraController.value.isRecordingVideo) {
      return;
    }

    try {
      await cameraController.pauseVideoRecording();
    } on CameraException catch (e) {
      _showCameraException(e);
      rethrow;
    }
  }

  Future<void> resumeVideoRecording() async {
    final CameraController? cameraController = controller;

    if (cameraController == null || !cameraController.value.isRecordingVideo) {
      return;
    }

    try {
      await cameraController.resumeVideoRecording();
    } on CameraException catch (e) {
      _showCameraException(e);
      rethrow;
    }
  }

  Future<void> setFlashMode(FlashMode mode) async {
    if (controller == null) {
      return;
    }

    try {
      await controller!.setFlashMode(mode);
    } on CameraException catch (e) {
      _showCameraException(e);
      rethrow;
    }
  }

  Future<void> setExposureMode(ExposureMode mode) async {
    if (controller == null) {
      return;
    }

    try {
      await controller!.setExposureMode(mode);
    } on CameraException catch (e) {
      _showCameraException(e);
      rethrow;
    }
  }

  Future<void> setExposureOffset(double offset) async {
    if (controller == null) {
      return;
    }

    setState(() {
      _currentExposureOffset = offset;
    });
    try {
      offset = await controller!.setExposureOffset(offset);
    } on CameraException catch (e) {
      _showCameraException(e);
      rethrow;
    }
  }

  Future<void> setFocusMode(FocusMode mode) async {
    if (controller == null) {
      return;
    }

    try {
      await controller!.setFocusMode(mode);
    } on CameraException catch (e) {
      _showCameraException(e);
      rethrow;
    }
  }

  Future<void> _startVideoPlayer() async {
    if (videoFile == null) {
      return;
    }

    final VideoPlayerController vController = kIsWeb
        ? VideoPlayerController.network(videoFile!.path)
        : VideoPlayerController.file(File(videoFile!.path));

    videoPlayerListener = () {
      if (videoController != null && videoController!.value.size != null) {
        // Refreshing the state to update video player with the correct ratio.
        if (mounted) {
          setState(() {});
        }
        videoController!.removeListener(videoPlayerListener!);
      }
    };
    vController.addListener(videoPlayerListener!);
    await vController.setLooping(true);
    await vController.initialize();
    await videoController?.dispose();
    if (mounted) {
      setState(() {
        imageFile = null;
        videoController = vController;
      });
    }
    await vController.play();
  }

  Future<XFile?> takePicture() async {
    final CameraController? cameraController = controller;
    if (cameraController == null || !cameraController.value.isInitialized) {
      showInSnackBar('Error: select a camera first.');
      return null;
    }

    if (cameraController.value.isTakingPicture) {
      // A capture is already pending, do nothing.
      return null;
    }

    try {
      final XFile file = await cameraController.takePicture();
      return file;
    } on CameraException catch (e) {
      _showCameraException(e);
      return null;
    }
  }

  void _showCameraException(CameraException e) {
    _logError(e.code, e.description);
    showInSnackBar('Error: ${e.code}\n${e.description}');
  }
}

/// CameraApp is the Main Application.
class CameraApp extends StatelessWidget {
  /// Default Constructor
  const CameraApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: CameraExampleHome(),
    );
  }
}

List<CameraDescription> _cameras = <CameraDescription>[];

Future<void> main() async {
  // Fetch the available cameras before initializing the app.
  try {
    WidgetsFlutterBinding.ensureInitialized();
    _cameras = await availableCameras();
  } on CameraException catch (e) {
    _logError(e.code, e.description);
  }
  runApp(const CameraApp());
}
```

**Comment by huycozy** [comment]
@JcmeLs I think this issue is caused by the implementation as the image stream has not been released yet. 
We can see the error message when starting to record a video: `A camera has started streaming images. startImageStream was called while a camera was streaming images.`.

This message is thrown from `isStreamingImages` check:

https://github.com/flutter/plugins/blob/910b61d1bf4cd89fd7c2db59659cfa2047f4859e/packages/camera/camera/lib/src/camera_controller.dart#L432-L437

So you can try calling `stopImageStream()` after `startImageStream`, for eg from [Android image streaming test](https://github.com/flutter/plugins/blob/main/packages/camera/camera/example/integration_test/camera_test.dart#L209-L244).

**Comment by JcmeLs** [comment]
Hi @huycozy 
I don't think it's “the image stream has not been released yet” issue.And I'm not calling startImageStream multiple times.I need the data of each frame for AI recognition, so this is a continuously on state. And the issue will only appear on some devices. Many devices are functional.Don't know if can find a Mi Ultra for testing?

**Comment by huycozy** [comment]
Hi @JcmeLs, 
Unfortunately, I don't have such a device. In my case with `Samsung Galaxy Tab A7 Lite`, the application was stopped while taking a picture, and an [error message in previous comment](https://github.com/flutter/flutter/issues/114862#issuecomment-1308245813) appeared when recording a video:

<details>
<summary>Crash logs when capturing</summary>

```bash
I/Camera  (17527): sending capture request
E/AndroidRuntime(17527): FATAL EXCEPTION: CameraBackground
E/AndroidRuntime(17527): Process: com.example.reproduce_issues, PID: 17527
E/AndroidRuntime(17527): java.lang.IllegalArgumentException: CaptureRequest contains unconfigured Input/Output Surface!
E/AndroidRuntime(17527): 	at android.hardware.camera2.CaptureRequest.convertSurfaceToStreamId(CaptureRequest.java:739)
E/AndroidRuntime(17527): 	at android.hardware.camera2.impl.CameraDeviceImpl.submitCaptureRequest(CameraDeviceImpl.java:1243)
E/AndroidRuntime(17527): 	at android.hardware.camera2.impl.CameraDeviceImpl.capture(CameraDeviceImpl.java:1115)
E/AndroidRuntime(17527): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.capture(CameraCaptureSessionImpl.java:189)
E/AndroidRuntime(17527): 	at io.flutter.plugins.camera.Camera.takePictureAfterPrecapture(Camera.java:644)
E/AndroidRuntime(17527): 	at io.flutter.plugins.camera.Camera.onConverged(Camera.java:228)
E/AndroidRuntime(17527): 	at io.flutter.plugins.camera.CameraCaptureCallback.process(CameraCaptureCallback.java:135)
E/AndroidRuntime(17527): 	at io.flutter.plugins.camera.CameraCaptureCallback.onCaptureCompleted(CameraCaptureCallback.java:169)
E/AndroidRuntime(17527): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl$1.lambda$onCaptureCompleted$3$CameraCaptureSessionImpl$1(CameraCaptureSessionImpl.java:693)
E/AndroidRuntime(17527): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl$1$$ExternalSyntheticLambda6.run(Unknown Source:8)
E/AndroidRuntime(17527): 	at android.os.Handler.handleCallback(Handler.java:938)
E/AndroidRuntime(17527): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(17527): 	at android.os.Looper.loopOnce(Looper.java:226)
E/AndroidRuntime(17527): 	at android.os.Looper.loop(Looper.java:313)
E/AndroidRuntime(17527): 	at android.os.HandlerThread.run(HandlerThread.java:67)
I/Process (17527): Sending signal. PID: 17527 SIG: 9
Lost connection to device.

```
</details>

<details>
<summary>Error message when recording video</summary>

```bash
I/flutter (18049): Error: A camera has started streaming images.
I/flutter (18049): Error Message: startVideoRecording was called while a camera was streaming images.
```
</details>

Labeling for further insights from the team. Sample code can be found at https://github.com/flutter/flutter/issues/114862#issuecomment-1308070834

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.3.8, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.3.8 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 52b3dc25f6 (23 hours ago), 2022-11-09 12:09:26 +0800
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 332079AF-E895-4685-910E-7B1E18B0C6B8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[!] Flutter (Channel master, 3.6.0-1.0.pre.11, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.6.0-1.0.pre.11 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e6fb12469d (9 minutes ago), 2022-11-09 21:41:49 -0500
    • Engine revision b42ed6933c
    • Dart version 2.19.0 (build 2.19.0-382.0.dev)
    • DevTools version 2.19.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 332079AF-E895-4685-910E-7B1E18B0C6B8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>

**Comment by JcmeLs** [comment]
hi @huycozy ，
I don't need video here, so I haven't encountered this error yet. But there is a problem on devices that can run Image Stream normally. Cannot take pictures when Image Stream is enabled. I don't know if you have encountered it.
In addition, some Xiaomi devices cannot use the Image Stream function normally. I don't know if there are any friends in the team who have relevant equipment to test. Or some cloud real device platforms can try?Thanks again for your help.

**Comment by huycozy** [comment]
Hi @JcmeLs 

>Cannot take pictures when Image Stream is enabled. I don't know if you have encountered it.

I also can not take pictures in this case, there was a crash as my comment at https://github.com/flutter/flutter/issues/114862#issuecomment-1309735096. I also try on my another device `Realme 6` but it crashes too (no crash log from the console). 
I will try to reproduce this issue if I can find a similar device. Also, the issue had the appropriate labels for the team to give more insights.

**Comment by JcmeLs** [comment]
@huycozy 
Thanks for your help. For “Image Stream issue” ,if possible, please try more Xiaomi devices. Easier to reproduce the problem.

**Comment by JcmeLs** [comment]
I just used XiaoMi 11 and had the same problem. After starting Image Stream, the camera screen will be stuck.

**Comment by felixriehm** [comment]
you could try https://pub.dev/packages/camerawesome

supports image stream via "image analysis"

this works for me where the camera plugin does not work

**Comment by camsim99** [comment]
This issue should be fixed in the [CameraX implementation](https://pub.dev/packages/camera_android_camerax) of the camera plugin, so please try it out!

If you find a similar issue with that implementation (or any other issues), please feel free to file a separate issue.

**Comment by felixriehm** [comment]
I have the [same problem](https://github.com/flutter/flutter/issues/125623). Now i have tested it with my Galaxy A04s. It works now. Thanks @camsim99!

It is a bit laggy / not smooth but thats maybe due to my device. There are lot of logs like this which might be the reason:

`I/.cameraxexample( 6717): Background concurrent copying GC freed 1785(217KB) AllocSpace objects, 38(27MB) LOS objects, 33% free, 23MB/35MB, paused 5.743ms,50us total 154.841ms`

**Comment by camsim99** [comment]
@felixriehm Can you elaborate on how what is laggy/not smooth and when you see those logs?

**Comment by felixriehm** [comment]
Running simple source code from [here](https://github.com/flutter/flutter/issues/125623) (shows what i mean, laggy):

https://github.com/flutter/flutter/assets/13718034/1b907b47-3625-47be-9f26-9af03e3f959a

Running the default camera app of my phone in video mode (no lag):

https://github.com/flutter/flutter/assets/13718034/aa84c896-b276-438c-aab1-c4ccd33943ac


Logs snippet:
```
...
I/.cameraxexample(25669): Background concurrent copying GC freed 2674(257KB) AllocSpace objects, 23(15MB) LOS objects, 49% free, 11MB/23MB, paused 5.952ms,51us total 41.747ms
I/.cameraxexample(25669): Background concurrent copying GC freed 2097(224KB) AllocSpace objects, 36(26MB) LOS objects, 38% free, 18MB/30MB, paused 6.532ms,166us total 57.505ms
E/BufferQueueProducer(25669): [ImageReader-1280x720f23m4-25669-1](id:644500000004,api:4,p:816,c:25669) waitForFreeSlotThenRelock: timeout [mDequeueTimeout=1000ms
...
```

**Comment by camsim99** [comment]
Separately, I wonder if this issue is related to https://github.com/flutter/flutter/issues/145893

**Comment by jamesstevensdev** [comment]
@camsim99  I have run into this issue as well. The camera stream will randomly stop on specific Android devices. Oddly, the controller.value.isStreamingImages still returns true despite there being no new images. I have been able to recreate the issue on these devices.

- Samsung A51 (sm-a156u)
- Xiaomi Poco F5 (23049pcd8g)
- Samsung s21 (never occurred while in debug mode, only release mode)

**Comment by camsim99** [comment]
@jamesstevensdev Oh interesting. And you're using the latest version of `android_camera_camerax`? Wondering if https://github.com/flutter/flutter/issues/145893 helps mitigate this issue or not.

**Comment by jamesstevensdev** [comment]
@camsim99 I am currently running on the following

  camera: ^0.10.5+9
  camera_android_camerax: ^0.6.3
  
0.6.3 already has the fix mentioned in #145893. I don't see the issue frequently, but it definitely happens on specific phones.

**Comment by jamesstevensdev** [comment]
@camsim99 Is there an estimated timeline for this bug fix?

**Comment by camsim99** [comment]
@jamesstevensdev This is not in our plans for the quarter, so I don't have one for you. If you have any specific debug information that might help me track down the problem more easily, I can try to find the bandwidth though.
